### ISSUE TITLE
Support Infinispan for http session cache

### ISSUE THREAD

**Issue by frowe** [OpenLiberty/open-liberty#8835]
RFE request for docker support of Infinispan for session cache: https://github.com/WASdev/ci.docker/issues/279
Feature design document (WAD): https://ibm.box.com/s/2qrcf5sp9l4v84bgx3b7iuai5u74f223
Since Infinispan implements the jCache API, it should be possible to configure Liberty to use it for http session caching. 
In addition to this basic support, we need to:
1) provide config snippets and directions similar to what we did for Hazelcast as documented at https://github.com/OpenLiberty/ci.docker#session-caching
2) update docker configure.sh for Infinispan
3) determine if the developer experience in terms of configuration can be improved 
4) verify this works properly in an OpenShift environment

Note: We won't be doing any value add to attempt to support connecting to an Infinispan server embedded in Liberty from another Liberty server configured as an Infinispan client.  It is likely this isn't even possible considering how the Infinispan embedded, client, and server appear to be very different config/binaries, and the embedded version likely doesn't support the necessary protocols to communicate.

**Comment by njr-11** [comment]
Serviceability Approval Comment - Please answer the following questions for serviceability approval:

    WAD -- does the WAD identify the most likely problems customers will see and identify how the feature will enable them to diagnose and solve those problems without resorting to raising a PMR? Have these issues been addressed in the implementation?
We cover this in the Skill Transfer, not the WAD. But otherwise, yes, the implementation addresses the known serviceability issues.

    Test and Demo -- As part of the serviceability process we're asking feature teams to test and analyze common problem paths for serviceability and demo those problem paths to someone not involved in the development of the feature (eg. L2, test team, or another development team).
    a) What problem paths were tested and demonstrated?
Various forms of misconfiguration of Infinispan for HTTP Session Persistence. Error/warning messages. Debug output that shows relevant portions of Infinispan config with passwords/sensitive info omitted. 
    b) Who did you demo to?
Greg Watts, James Stephens on Thursday, Nov 21.
    c) Do the people you demo'd to agree that the serviceability of the demonstrated problem scenarios is sufficient to avoid PMRs for any problems customers are likely to encounter, or that L2 should be able to quickly address those problems without need to engage L3?
yes

    SVT -- SVT team is often the first team to try new features and often encounters problems setting up and using them. Note that we're not expecting SVT to do full serviceability testing -- just to sign-off on the serviceability of the problem paths they encountered.
    a) Who conducted SVT tests for this feature?
Nichole Stewart
    b) Do they agree that the serviceability of the problems they encountered is sufficient to avoid PMRs, or that L2 should be able to quickly address those problems without need to engage L3?
Need to wait for SVT to be completed before we can expect them to answer this.  To be updated once we have the answer.

    Which L2 / L3 queues will handle PMRs for this feature? Ensure they are present in the contact reference file and in the queue contact summary, and that the respective L2/L3 teams know they are supporting it. Ask Don Bourne if you need links or more info.
WAS: L3 Sessions
WAS: L2 Web
Allan Zhang is aware of this feature and that he will be supporting it alongside all other configurations of HTTP Session Persistence

**Comment by atosak** [comment]
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## List of Steps to complete or get approvals / sign-offs for Onboarding to the Liberty release (GM date)
 
Instructions:
 - Do the actions below and mark them complete in the checklist when they are done.
 - Make sure all feature readiness approvers put the appropriate tag on the epic to indicate their approval.
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## TARGET COMPLETION DATE **Before Development Starts or 8 weeks before Onboarding** 
- [ ] POC Design / WAD Review Scheduled (David Chang) or N/A.
- [ ] POC Design / WAD Reviewed (Feature Owner) or N/A.
- [ ] Complete any follow-ons from the POC Review.
- [ ] Design / WAD Approval (Alasdair Nottingham) or N/A.
- [ ] No Design / No WAD Approval (Arthur De Magalhaes - cloud / Alasdair Nottingham - server) or N/A.
- [ ] SVT Requirements identified. (Epic owner / Feature owner with SVT focal point)
- [ ] ID Requirements identified. (Epic owner / Feature owner with ID focal point)
- [ ] Create a child task of the epic entitled "FAT Approval Test Summary".  Add and fill in the template as described here: https://github.ibm.com/was-liberty/WS-CD-Open/wiki/Feature-Review-(Feature-Test-Summary-Process)
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## TARGET COMPLETION DATE **3 weeks before Onboarding** 
- [ ] Identify all open source libraries that are changing or are new. Work with Legal Release Services (Cass Tucker or Release PM) to get open source cleared and approved. Or N/A. (Epic Owner).   New or changed open source impacts license and Certificate of Originality.
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## TARGET COMPLETION DATE ** 3 weeks before Onboarding** 
- [ ] All new or changed PII messages are checked into the integration branch, before the last translation shipment out. (Epic Owner)
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## TARGET COMPLETION DATE **2 weeks before Onboarding** 
- [ ] Implementation complete. (Epic owner / Feature owner) 
- [ ] All function tests complete. Ready for FAT Approval. (Epic owner / Feature owner)
- [ ] Review all known issues for Stop Ship. (Epic owner / Feature owner / PM)
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## APPROVALS with TARGET COMPLETION DATE **2 to 1 week before Onboarding** 
Prereq: You must have the Design Approved or No Design Approved label on the GitHub Epic.
- [ ] **Accessibility** - (G Scott Johnston). Accessibility testing is complete or N/A. Approver adds label focalApproved:accessibility to the Epic in Github.
- [ ] **FAT Liberty SOE** - (Kevin Smith). SOE FATS are running successfully or N/A . Approver adds label focalApproved:fat to the Epic in Github.
- [ ] **Globalization** (Sam Wong - Liberty / Simy Cheeran - tWAS). Translation is complete or N/A. TVT - complete or N/A. Approver adds label focalApproved:globalization to the Epic in Github.
- [ ] **ID** - (Kareen Deen). Documentation work is complete or N/A . Approver adds label focalApproved:id to the Epic in Github.
- [ ] **Performance** - (Jared Anderson). Performance testing is complete with no high severity defects or N/A . Approver adds label focalApproved:performance to the Epic in Github.
- [ ] **Serviceability** - (Don Bourne). Serviceability has been addressed. 
- [ ] **STE** - (Swati Kasundra). STE chart deck is complete or N/A . Approver adds label focalApproved:ste to the Epic in Github.
- [ ] **SVT** - (Greg Ecock - Cloud, Brian Hanczaryk- APS). SVT is complete or N/A . Approver adds label focalApproved:svt to the Epic in Github.
- [ ] **Demo** - (Liberty only - Tom Evans or Chuck Bridgham). Demo is scheduled for an upcoming EOI. Approver adds label focalApproved:demo to the Epic in Github.
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
## TARGET COMPLETION DATE **1 week before Onboarding** 
- [ ] No Stop Ship issues for the feature. (Epic owner / Feature owner / Release PM)
- [ ] Ship Readiness Review and Release Notes completed (Epic owner / Feature owner / Release PM)
- [ ] Github Epic and Epic's issues are closed / complete. All PRs are committed to the master branch. (Epic owner / Feature owner / Backlog Subtribe PM)
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## NOT REQUIRED FOR A FEATURE
- [ ] **OL Guides** - (Yee-Kang Chang). Assessment for OL Guides is complete or N/A. 
- [ ] **WDT** - (Leonard Theivendra). WDT work complete or N/A.
 
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
## Related Deliverables TARGET COMPLETION DATE **General Availability** 
- [ ] Blog article writeup (Epic owner / Feature owner / Laura Cowen)

**Comment by gscottj** [comment]
This feature does not add any new or change existing user interfaces in ways that would require additional accessibility verification testing. This feature only involve adding standard metatype config.

**Comment by skasund** [comment]
L2 has requested STE slides for this feature.  The STE template can be found at the link below.  You can use either one to create the education.  

Slide Template:  https://ibm.box.com/s/1an42g7zdgmaj84w7dft0indqfgi8ffm

Github Template:  https://pages.github.ibm.com/WASL3/site/STE/about

Please upload the completed slides to the same STE Archive BOX folder.  Thanks!

**Comment by chirp1** [comment]
Fred indicated that no ID is required. Approving.

**Comment by donbourne** [comment]
granting serviceability approval with understanding that @nstewart0206 will withhold SVT approval if there are any serviceability issues found.
